File: flutter17_mobile/lib/models/accessory.dart
import 'package:flutter17_mobile/models/accessory_category.dart';
import 'package:flutter17_mobile/models/accessory_property.dart';
import 'package:json_annotation/json_annotation.dart';

part 'accessory.g.dart';

@JsonSerializable(explicitToJson: true)
class Accessory {
  int? id;
  String? name;
  String? description;
  int? productId;
  AccessoryCategory? accessoryCategory;
  List<AccessoryProperty>? accessoryProperties;

  Accessory({
    this.id,
    this.name,
    this.description,
    this.productId,
    this.accessoryCategory,
    this.accessoryProperties,
  });

  factory Accessory.fromJson(Map<String, dynamic> json) => _$AccessoryFromJson(json);

  Map<String, dynamic> toJson() => _$AccessoryToJson(this);
}

File: angular17_web/src/app/models/accessory/accessory.ts
import {AccessoryCategory} from '../accessory-category/accessory.category';
import {AccessoryProperty} from '../accessory-property/accessory.property';

export class Accessory {
  id!: number;
  name!: string;
  description!: string;
  productId!: number;
  accessoryCategory?: AccessoryCategory;
  accessoryProperties: AccessoryProperty[] = [];

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.name = data.name;
    this.description = data.description;
    this.productId = data.productId;

    this.accessoryCategory = data.accessoryCategory
     ? new AccessoryCategory(data.accessoryCategory)
     : undefined;
    this.accessoryProperties = Array.isArray(data.accessoryProperties)
     ? data.accessoryProperties.map((x: any) => new AccessoryProperty(x))
     : [];
  }
}
